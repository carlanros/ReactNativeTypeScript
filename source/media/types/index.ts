// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
