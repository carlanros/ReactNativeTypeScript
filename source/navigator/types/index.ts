export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
