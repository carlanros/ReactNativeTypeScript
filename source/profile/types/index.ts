// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
