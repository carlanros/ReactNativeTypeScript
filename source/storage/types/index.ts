// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}