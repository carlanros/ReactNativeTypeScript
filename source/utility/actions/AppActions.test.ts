// utility/actions/AppActions.test
/// <reference types="jest"/>
// @ts-ignore - bad mappings
import configureMockStore from 'redux-mock-store';

import { Store } from '../../Store';
import * as types from '../types/';
import { AppLoadingChanged, AppErrorChanged } from './';

/**
 * Mock Redux store
 */
const store = configureMockStore<Store>();

// Navigator will call connected components
jest.mock('../../navigator', () => ({
  Routes: 'View',
  AppNavigator: 'Button',
}));

describe('App actions', () => {
  beforeEach(() => {
    // @ts-ignore - bad mappings
    store.clearActions();
  });

  test('AppLoadingChanged dispatches busy reason', () => {
    const busyReason = 'Loading test';

    const expectedActions = [
      {
        type: types.UtilityConstants.APP_LOAD_BUSY,
        reason: busyReason,
        sender: 'TestMethod',
      },
    ];

    // Dispatch action
    // @ts-ignore - bad mappings
    store.dispatch(
      AppLoadingChanged({
        isBusy: true,
        reason: busyReason,
        sender: 'TestMethod',
      })
    );

    // @ts-ignore - bad mappings
    expect(store.getActions()).toMatchSnapshot();
    // @ts-ignore - bad mappings
    expect(store.getActions()).toEqual(expectedActions);
  });

  test('AppErrorChanged dispatches error message', () => {
    const errorMessage = 'Error test';

    const expectedAction: types.IAppErrorChanged = {
      type: types.UtilityConstants.APP_HAS_ERROR,
      reason: errorMessage,
      exception: null,
      sender: null,
    };

    const expectedActions = [expectedAction];

    // Dispatch action
    // @ts-ignore - bad mappings
    store.dispatch(
      AppErrorChanged({
        hasError: true,
        reason: errorMessage,
        sender: null,
      })
    );

    // @ts-ignore - bad mappings
    expect(store.getActions()).toMatchSnapshot();
    // @ts-ignore - bad mappings
    expect(store.getActions()).toEqual(expectedActions);
  });
});
