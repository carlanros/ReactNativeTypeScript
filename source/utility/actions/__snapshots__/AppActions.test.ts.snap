// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App actions AppErrorChanged dispatches error message 1`] = `
Array [
  Object {
    "exception": null,
    "reason": "Error test",
    "sender": null,
    "type": "APP = HAS ERROR",
  },
]
`;

exports[`App actions AppLoadingChanged dispatches busy reason 1`] = `
Array [
  Object {
    "reason": "Loading test",
    "sender": "TestMethod",
    "type": "APP = BUSY LOADING",
  },
]
`;
